vangoh.go

```diff
@@ -247,7 +254,14 @@ func (vg *Vangoh) AuthenticateRequest(r *http.Request) *AuthenticationError {
 	actualSignatureB64 := keySplit[1]
 
 	// Check that the request was made in the acceptable window.
-	dateHeader := strings.TrimSpace(r.Header.Get("Date"))
+	dateHeader := ""
+	if vg.customDateHeader != "" {
+		dateHeader = strings.TrimSpace(r.Header.Get(vg.customDateHeader))
+	}
+	//fallback on "Date" header
```

Comment:
Is falling back the desired behavior? E.g., In the event that the server is configured to use the custom date header `X-AWS-Date:` and the client sends a normal `Date:` header, would it make more sense for the server to fail instead of silently succeeding? I guess I'm thinking about past experience working with other teams integrating against backend code I've written, it's usually best if there's only one possible way for them to integrate correctly.